server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestResetFeatureStateAction.java

```diff
@@ -39,6 +41,19 @@ public String getName() {
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         final ResetFeatureStateRequest req = new ResetFeatureStateRequest();
 
-        return restChannel -> client.execute(ResetFeatureStateAction.INSTANCE, req, new RestToXContentListener<>(restChannel));
+        return restChannel -> client.execute(
+            ResetFeatureStateAction.INSTANCE,
+            req,
+            new RestToXContentListener<>(restChannel) {
+                @Override
+                protected RestStatus getStatus(ResetFeatureStateResponse response) {
+                    if (response.hasAllFailures()) {
```

Comment:
So I've got a 500 if all resets fail, a 207 for mixed responses, and a 200 if all goes well. Does this make sense?